src/plugins/lua/gpt.lua

```diff
@@ -683,10 +683,54 @@ local function openai_check(task, content, sel_part)
 
   local from_content, url_content = get_meta_llm_content(task)
 
-  local body = {
-    model = settings.model,
-    max_tokens = settings.max_tokens,
-    temperature = settings.temperature,
+  -- Deep copy to avoid table reuse between iterations
+  local function table_deep_copy(orig)
+    local copy
+    if type(orig) == 'table' then
+      copy = {}
+      for k, v in pairs(orig) do
+        copy[k] = table_deep_copy(v)
+      end
+    else
+      copy = orig
+    end
+    return copy
+  end
+
+  -- Decide which token length field to use for the given model
+  local function get_max_tokens_field(model)
+    if not model then
+      return 'max_tokens' -- default
+    end
+
+    -- Newer models that require max_completion_tokens
+    if model:match('^gpt%-5') or
+       model:match('^o%d') or
+       model:match('^o%d%-mini') or
+       model:match('^gpt%-4%.1') or
+       model:match('reasoning') then
+      return 'max_completion_tokens'
+    end
+
+    -- Default for older/legacy models
+    return 'max_tokens'
+  end
+
+  -- Only send temperature if model supports it
+  local function supports_temperature(model)
+    if not model then return true end
+    -- Disallow for reasoning models and GPT-5 family
+    if model:match('^gpt%-5') or
+       model:match('^o%d') or
+       model:match('^o%d%-mini') or
+       model:match('^gpt%-4%.1') or
+       model:match('reasoning') then
```

Comment:
I don't like this approach. We need to create either config option or a table with something like `model` -> `parameters` where we can set it. Matching 5 regular expressions is very far from optimal approach.

Reply:
I've made a change to get the model parameters from the settings now. Due this change the documentation needs to be updated.